src/OversetSimulation.cpp

```diff
@@ -138,15 +140,18 @@ void OversetSimulation::run_timesteps(const int add_pic_its, const int nsteps)
 
         m_timers_exa.tick("TimeStep");
 
-        for (auto& ss : m_solvers) ss->call_pre_advance_stage1();
+        for (size_t inonlin=0; inonlin < nonlinear_its; inonlin++) {
+            
+            for (auto& ss : m_solvers) ss->call_pre_advance_stage1(inonlin);
 
-        if (do_connectivity(nt)) perform_overset_connectivity();
+            if (do_connectivity(nt)) perform_overset_connectivity();
+            
+            for (auto& ss : m_solvers) ss->call_pre_advance_stage2(inonlin);
 
-        for (auto& ss : m_solvers) ss->call_pre_advance_stage2();
-
-        exchange_solution();
+            exchange_solution();
```

Comment:
change to 
`increment_timer = inonlin > 0 ? true : false`
`exchange_solution(increment_timer)`

Reply:
Once this PR is merged, I will add the timer increments to other methods under the nonlinear iteration loop.